### DISCUSSION TITLE
Requiring bash coding style

### DISCUSSION THREAD

**Discussion by orazioedoardo** [pivpn/pivpn#1573] [Feature requests]
Over the years several people have contributed to the projects will many different coding styles, sometimes updating other people formatting and also messing with indentation in the process (I did it for example). I think it's time to require a consistent coding style, like this one from google: https://google.github.io/styleguide/shellguide.html

My plan:

- Make a decision
- Update all scripts in the project to fit the style
- Start requiring new pull request to adhere to the coding style.

Tagging active users on the project (non exhaustive) @4s3ti @MichaIng @TinCanTech, also @DerDanilo and @giulio-coa

**Comment by ghost** [comment]
I think it is a good idea and I approve the proposed code style.
I also propose to require the sign-off of the commits.

**Comment by MichaIng** [comment]
Agreed, the more people contribute, the more a coding style makes sense.

Does someone know a good console tool to check shell coding style? `shellcheck` only checks syntax, common mistakes or performance related stuff. Would be great to have a GitHub actions workflow for this.

EDIT, probably: https://github.com/mvdan/sh/blob/master/cmd/shfmt/shfmt.1.scd

**Reply by coolapso** [reply]
I am familiar with shfmt .. and can implement it on CI environment, not 100% if it fully covers the Google bash code style, but i think it covers most of it.

**Comment by coolapso** [comment]
I am happy with that code style, and as stated in the comments above i think we can use shfmt to do the checks.

**Reply by ghost** [reply]
> I am happy with that code style, and as stated in the comments above i think we can use shfmt to do the checks.

https://github.com/luizm/action-sh-checker

**Comment by ghost** [comment]
If you implement it into the CI enviroment and merge my PR, I merge the changes with my `formatting_enhancements`  branch and adapt it to the new code style rules

**Reply by coolapso** [reply]
Lets not rush, we do the Alpine first.
Then we do the code style in separate PR's.

**Reply by ghost** [reply]
> Lets not rush, we do the Alpine first.
> Then we do the code style in separate PR's. 

Yep
I have already the PR
I'm only waiting the merge

**Comment by ghost** [comment]
[#1572](https://github.com/pivpn/pivpn/pull/1572)
